presto-native-execution/presto_cpp/main/PrestoTask.cpp

```diff
@@ -117,45 +118,126 @@ static void addRuntimeMetric(
   }
 }
 
-// Updates a Velox runtime metric in the unordered map if the value is not 0.
-static void addRuntimeMetricIfNotZero(
-    std::unordered_map<std::string, RuntimeMetric>& runtimeMetrics,
-    const std::string& name,
-    uint64_t value) {
-  if (value > 0) {
-    auto veloxMetric =
-        createVeloxRuntimeMetric(value, RuntimeCounter::Unit::kNone);
-    addRuntimeMetric(runtimeMetrics, name, veloxMetric);
-  }
-}
-
 // Add 'spilling' metrics from Velox operator stats to Presto operator stats.
 static void addSpillingOperatorMetrics(
     protocol::OperatorStats& opOut,
-    protocol::TaskStats& prestoTaskStats,
     const exec::OperatorStats& op) {
-  std::string statName =
-      fmt::format("{}.{}.spilledBytes", op.operatorType, op.planNodeId);
-  auto prestoMetric = createProtocolRuntimeMetric(
-      statName, op.spilledBytes, protocol::RuntimeUnit::BYTE);
-  opOut.runtimeStats.emplace(statName, prestoMetric);
-  prestoTaskStats.runtimeStats[statName] = prestoMetric;
-
-  statName = fmt::format("{}.{}.spilledRows", op.operatorType, op.planNodeId);
-  prestoMetric = createProtocolRuntimeMetric(statName, op.spilledRows);
-  opOut.runtimeStats.emplace(statName, prestoMetric);
-  prestoTaskStats.runtimeStats[statName] = prestoMetric;
-
-  statName =
-      fmt::format("{}.{}.spilledPartitions", op.operatorType, op.planNodeId);
-  prestoMetric = createProtocolRuntimeMetric(statName, op.spilledPartitions);
-  opOut.runtimeStats.emplace(statName, prestoMetric);
-  prestoTaskStats.runtimeStats[statName] = prestoMetric;
-
-  statName = fmt::format("{}.{}.spilledFiles", op.operatorType, op.planNodeId);
-  prestoMetric = createProtocolRuntimeMetric(statName, op.spilledFiles);
-  opOut.runtimeStats.emplace(statName, prestoMetric);
-  prestoTaskStats.runtimeStats[statName] = prestoMetric;
+  if (op.spilledBytes == 0) {
+    return;
+  }
+
+  const std::vector<std::tuple<std::string, uint64_t, protocol::RuntimeUnit>>
+      spillMetrics{
+          std::make_tuple(
+              "spilledBytes", op.spilledBytes, protocol::RuntimeUnit::BYTE),
+          std::make_tuple(
+              "spilledRows", op.spilledRows, protocol::RuntimeUnit::NONE),
+          std::make_tuple(
+              "spilledPartitions",
+              op.spilledPartitions,
+              protocol::RuntimeUnit::NONE),
+          std::make_tuple(
+              "spilledFiles", op.spilledFiles, protocol::RuntimeUnit::NONE)};
+
+  for (const auto& spillMetric : spillMetrics) {
+    std::string statName = fmt::format(
+        "{}.{}.{}", op.operatorType, op.planNodeId, std::get<0>(spillMetric));
+    opOut.runtimeStats.emplace(
+        statName,
+        createProtocolRuntimeMetric(
+            statName, std::get<1>(spillMetric), std::get<2>(spillMetric)));
+  }
+}
+
+static void addTaskRuntimeMetrics(
```

Comment:
The name is quite generic.
Do I understand correctly that this function appends three operator metrics ("numSplits", etc.) from operator stats to the collection of "would be" task level runtime metrics?
If so, we should reflect this in the name. 3 metrics sound like "inputOutputStats"...

Also, what if these numbers are zero and not applicable to the metric?
The old code was checking each of of them for != 0...